### PR TITLE
Don't trigger onMouseOver/onMouseOut events when relatedTarget is child of currentTarget

### PR DESCRIPTION
**Pull request by samanpwbb**
This PR fixes an issue where onMouseOver/onMouseOut get called repeatedly even if you do not exit a particular square _if_ that square has a piece on it. This occurs because the cursor moves from the square into the piece. If there is no piece on the square, these events only get called on enter and exit. I think it would be more predictable to always only trigger the callbacks when actually leaving and entering squares. 

This PR also slightly restructures the repo. I did this to make the repo installable from github paths  (for ex: `"react-chessboard": "samanpwbb/react-chessboard#fix-events"`). I can revert this change if you'd like, but I don't see the downsides. It would not be a breaking change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
+  "main": "package/dist/index.js",
+  "module": "package/dist/index.esm.js",
+  "types": "package/dist/index.d.ts",
   "files": [
-    "dist"
+    "package/dist"
   ],
   "scripts": {
-    "build": "cp ../README.md . && rollup -c",
+    "build": "rollup -c",
```
Comment (Clariity) [inline package.json:29]: the readme still needs to be copied from the root into the package for when it's published to npm

**Comment by Clariity** [comment]
build is failing because the install path in the example is now pointing to the wrong directory, can that be fixed please and then I can approve

great fix for the issue, thank you for the contribution

**Comment by samanpwbb** [comment]
Sweet, I'll get the git history cleaned up and address your feedback soon 👍

File: package.json (hunk 1 above)
Comment (samanpwbb) [reply package.json:29]: Will do!

**Pull request closed without merging**